tensorflow/python/ops/control_flow_ops.py

```diff
@@ -3611,12 +3611,23 @@ def _case_verify_and_canonicalize_args(pred_fn_pairs, exclusive, name,
   if isinstance(pred_fn_pairs, collections.OrderedDict):
     pred_fn_pairs = pred_fn_pairs.items()
   elif isinstance(pred_fn_pairs, dict):
-    pred_fn_pairs = sorted(pred_fn_pairs.items(), key=lambda item: item[0].name)
-    if not exclusive:
-      logging.warn(
-          "%s: An unordered dictionary of predicate/fn pairs was "
-          "provided, but exclusive=False. The order of conditional "
-          "tests is deterministic but not guaranteed.", name)
+    if context.executing_eagerly():
+      # No name to sort on in eager mode. Use dictionary traversal order, 
+      # which is nondeterministic in versions of Python < 3.6
+      if not exclusive and not isinstance(pred_fn_pairs, 
```

Comment:
Minor nit: I don't think you need the "isinstance(pred_fn_pairs, collections.OrderedDict)".

Reply:
Oops! I didn't notice the if statement 4 lines up :-\ Removed that predicate.